### PR TITLE
Feature/ti2023 39/frontend functionality

### PR DESCRIPTION
**Pull request by Filip-B-32**

### CODE REVIEW

File: EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace EventPlannerAPI.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
-    [Authorize(Roles = RoleConstants.ADMIN_ROLE)]
+    //[Authorize(Roles = RoleConstants.ADMIN_ROLE)]
```
Comment (alexandratodea) [inline EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs]: commented code shouldn't be committed

File: EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs [hunk 2]
```diff
@@ -133,7 +133,7 @@ public async Task<ActionResult> UpdateEvent(Guid eventId, [FromBody] UpdateEvent
             }
         }
 
-        [Authorize]
+        //[Authorize]
```
Comment (alexandratodea) [inline EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs]: commented code shouldn't be committed

File: EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs (hunk 1 above)
Comment (gheorghe-sava) [inline EventPlannerAPI/EventPlannerAPI/Controllers/AdminController.cs]: Why was this commented?

File: EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs (hunk 2 above)
Comment (gheorghe-sava) [inline EventPlannerAPI/EventPlannerAPI/Controllers/EventController.cs]: Why was this commented?

File: event-planner-frontend/src/features/browse-events/components/events-page/events-card-list/join-modal/JoinModal.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    const data: JoinEventDto = {
+      userId: userId,
+      eventId: eventId,
+      eventPositionId: selectedPosition,
+    };
+    await dispatch(joinEventThunk(data));
+    onJoinSuccess();
+    setTimeout(() => {
+      window.location.reload();
+    }, 2000);
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/browse-events/components/events-page/events-card-list/join-modal/JoinModal.tsx]: Why do we need a setTimeout?

File: event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+const initialState: State = {
+  isLoading: false,
+  isSuccess: false,
+  isDone: false,
+  error: '',
+};
+
+const joinEventSlice = createSlice({
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts]: Creating a slice for each action is too tedious to work with and to manage, a slice per module would be enough.

Please use the existing slice.

File: event-planner-frontend/src/features/browse-events/thunks/joinEventsThunk.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import EventService from 'services/eventService';
+import { JoinEventDto } from '../api/dtos';
+import { handleApiError, handleGenericSuccess } from 'services/notificationHandlingService';
+
+const eventService: EventService = new EventService();
+
+export const joinEventThunk = createAsyncThunk('post/joinEvent', async (data: JoinEventDto, { rejectWithValue }) => {
```
Comment (gheorghe-sava) [inline event-planner-frontend/src/features/browse-events/thunks/joinEventsThunk.ts]:
```suggestion
export const joinEventThunk = createAsyncThunk('eventsSlice/joinEvent', async (data: JoinEventDto, { rejectWithValue }) => {
```

File: event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts (hunk 4 above)
Comment (Filip-B-32) [reply event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts]: you're right but i made a different thunk and slice for each api. The browse events api is different than the join one

Comment (gheorghe-sava) [reply event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts]: It doesn't matter, it gets too complex to manage this many slices
Unless there is a valid reason to create separate slices like different features, we can reuse existing slices

Comment (gheorghe-sava) [reply event-planner-frontend/src/features/browse-events/store/slices/joinEventSlice.ts]: The same applies to thunks, we can create more thunks in the same file if the number of thunks is not that large

File: event-planner-frontend/src/features/browse-events/components/events-page/events-card-list/join-modal/JoinModal.tsx (hunk 3 above)
Comment (Filip-B-32) [reply event-planner-frontend/src/features/browse-events/components/events-page/events-card-list/join-modal/JoinModal.tsx]: after 2 seconds (after the toast has been displayed) it should reload the page with the updated event (when i join an event the certain position and maximum positions are going to be less with one)

**Review by gheorghe-sava — APPROVED** [review]

**Review by alexandratodea — APPROVED** [review]

**Pull request merged**